MapboxDirections/MBLane.swift

```diff
@@ -0,0 +1,83 @@
+import Foundation
+
+public enum LaneIndicationType: Int, CustomStringConvertible {
+    case Left
+    
+    case Right
+    
+    case SharpLeft
+    
+    case SharpRight
+    
+    case SlightLeft
+    
+    case SlightRight
+    
+    case Straight
+    
+    case Uturn
+    
+    public init?(description: String) {
+        let type: LaneIndicationType
+        switch description {
+        case "left":
+            type = .Left
+        case "right":
+            type = .Right
+        case "sharp left":
+            type = .SharpLeft
+        case "sharp right":
+            type = .SharpRight
+        case "slight left":
+            type = .SlightLeft
+        case "slight right":
+            type = .SlightRight
+        case "straight":
+            type = .Straight
+        case "uturn":
+            type = .Uturn
+        default:
+            return nil
+        }
+        self.init(rawValue: type.rawValue)
+    }
+    
+    public var description: String {
+        switch self {
+        case .Left:
+            return "left"
+        case .Right:
+            return "right"
+        case .SharpLeft:
+            return "sharp left"
+        case .SharpRight:
+            return "sharp right"
+        case .SlightLeft:
+            return "slight left"
+        case .SlightRight:
+            return "slight right"
+        case .Straight:
+            return "straight"
+        case .Uturn:
+            return "uturn"
+        }
+    }
+}
+
+public class Lane: NSObject {
+    public var validTurn: Bool
```

Comment:
I know that’s how it’s exposed in the API. But the API doesn’t use classes at all, so that’s comparing apples and oranges.

When you create a class, you make it possible for the developer to have an instance of that class all by itself. If the developer has a Lane object but not the Intersection or RouteStep object that contains it, `validTurn` is meaningless or misleading. It’s saying that a physical lane on a road can always or never be used for all routes that may contain the lane, which is unrealistic.

Semantically, the information you’re trying to store must live in a parallel data structure. Specifically, Intersection should have a `usableLanes` property that _contains_ lanes or lane indices, and this `validTurn` property should be removed.